native/src/main/java/io/ballerina/lib/data/xmldata/xml/XSDValidator.java

```diff
@@ -78,7 +78,9 @@ private static Object validateXmlFromXsdFile(String xsdFilePath, BXml xml)
             throws ParserConfigurationException, IOException, SAXException {
         try {
             DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            dbFactory.setNamespaceAware(true); // Support for XML namespaces
             dbFactory.setIgnoringComments(true); // Ignore comments in the XML
+            dbFactory.setIgnoringElementContentWhitespace(true); // Ignore white spaces in the XML
```

Comment:
We may need to provide a configuration to override the removal of leading and trailing whitespaces

Reply:
I will revert that